src/main/java/org/folio/copycat/RecordRetriever.java

```diff
@@ -89,11 +89,28 @@ static void setAuthOptions(Connection conn, String auth) {
       ResultSet search = conn.search(query);
       Record record = search.getRecord(0);
       if (record == null) {
-        return Future.failedFuture("No record found");
+        return Future.failedFuture("No record found when searching "
+          + profile.getUrl() + " for identifier " + externalId);
       }
       return Future.succeededFuture(record.get(type));
     } catch (ZoomException e) {
-      return Future.failedFuture(e);
+      /*
+       https://github.com/indexdata/yaz4j/blob/master/src/main/java/org/yaz4j/ExceptionUtil.java
+       some messages include the server URL , some don't.. No sub classes for the exception
+       so we can determine if this is init or other error, and no way to get the error code.
+       Using this hack to determine if this was init error.
+       */
+      if (e.getMessage().endsWith(" init request")) {
+        return Future.failedFuture("Z39.50 error: server " + profile.getUrl() + " rejected init."
+          + " This is probably due to missing or incorrect authentication for the copycat profile");
```

Comment:
I would prefer the wording "This may be due to ...", but if the current version came from a PO I guess we should leave it as it is.

Reply:
Sure. I'll change it. Question: this is one long string (with no newline).. Will the UI break it up nicely?

Reply:
In general, these strings end up embedded in HTML in the UI, so they get word-wrapped in the standard way. So, yes, all should be well.